assignments/homework/pcosta/497HW2.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.io import fits
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.table import QTable
+from astropy.table import Table
+
+#Notebook 4
+#1
+gaiaLMC = fits.open('../../../astropy_notebooks/data/gaia_lmc_psc.fits')
+firstHDUHeader = gaiaLMC[0].header
+secondHDUHeader = gaiaLMC[1].header
+print(firstHDUHeader)
+print(secondHDUHeader)
+firstHDUHeader['BIGFISH'] = 'Always.'
+secondHDUHeader['TWO'] = "No more, no less."
+
+#2
+data = gaiaLMC[1].data
+gBandMag = data['phot_g_mean_mag']
+num_bins = 25
+n, bins, patches = plt.hist(gBandMag, num_bins, facecolor='blue')
+plt.show()
+maxGmag = np.amax(data['phot_g_mean_mag'])
+print("Upper limit for Gmag:",maxGmag)
+
+#3
+plotCoords = SkyCoord(data['ra']*u.deg, data['dec']*u.deg, frame='fk5')
+plotGalactic = plotCoords.transform_to('galactic')
+lCoord = plotGalactic.l
+bCoord = plotGalactic.b
+plt.plot(lCoord, bCoord, 'o', markersize=2)
+plt.xlabel("l Coordinate (Degrees)")
+plt.ylabel("b coordinate (Degrees)")
+plt.show()
+
+#4
+hduImage = fits.ImageHDU(fits.open('../../../astropy_notebooks/data/LMCDensFits1k.fits'))
+hduTable = fits.BinTableHDU(fits.open('../../../astropy_notebooks/data/gaia_lmc_psc.fits'))
+hduImage.writeto('myFits.fits')
+hduTable.writeto('myFits.fits')
```

Comment:
This solution doesn't work for several reasons:

- you cannot initialize either an ImageHDU nor a BinTableHDU with a HDUList. However, there is also no need to create those objects: edit the first two lines to open the fits files and keep one extension of them by indexing. For the image fits file it's the `0`, for the table it's the `1` extension.
Then you create a new `HDUList`, with specifying a list of these two HDUs, and write it out into a file. Currently you would overwrite the same file that still doesn't have both extensions.